dagsync/ipnisync/publisher.go

```
@@ -218,7 +223,14 @@ func (p *Publisher) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "invalid request: not a cid", http.StatusBadRequest)
 		return
 	}
-	item, err := p.lsys.Load(ipld.LinkContext{}, cidlink.Link{Cid: c}, basicnode.Prototype.Any)
+
+	ipldCtx := ipld.LinkContext{}
+	reqType := r.Header.Get(CidSchemaHeader)
+	if reqType != "" {
+		ipldCtx.Ctx = context.WithValue(context.Background(), CidSchemaCtxKey, reqType)
+	}
+
+	item, err := p.lsys.Load(ipldCtx, cidlink.Link{Cid: c}, basicnode.Prototype.Any)
```

Why `basicnode.Prototype.Any` if type is set? We probably should return an error if type is set in request and it doesn't match the type we read from link system.